src/utils/logger.cpp

```diff
@@ -1,16 +1,22 @@
-#include <memory>
-
 #include "logger.h"
+#include <memory>
+#include <stdlib.h>
 
 /**
  * \brief Logger implementation based on spdlog
  */
 struct Logger {
     logger_t logger;
     Logger(const std::string& name, std::string pattern) {
+        // Ensure settings are loaded from ENV variables
+        spdlog::cfg::load_env_levels();
+
+        // Create the logger and configure it
         logger = spdlog::stdout_color_mt(name);
         logger->set_pattern(std::move(pattern));
-        logger->set_level(spdlog::level::info);
+        if (getenv("SPDLOG_LEVEL") == NULL) {
```

Comment:
Just a nitpick for future PR - env variables with library specific would have been better. for example, this is sonata report lib specific. So something like `SONATA_REPORT_SPDLOG_LEVEL`

Reply:
This variable is coming from spdlog

Reply:
@pramodk This is just ensuring that the user did not ask for the level manually. [The ENV variable here comes from the `spdlog` implementation](https://github.com/gabime/spdlog#load-log-levels-from-env-variable-or-from-argv). If the user set a level from outside for `spdlog`, it is configured internally appropriately and the if-statement is to avoid setting the debug level manually by ourselves.

We did not want to have our own ENV variable and tried to followed the indications by `spdlog`, but it can be changed if you prefer.

What do you think?

Reply:
Didn't know they have "standard" `SPDLOG_LEVE` var. All good!